### PR TITLE
make first release ever through CI possible

### PR DESCRIPTION
**Pull request by bottlenecked**
I attempted to make my first release to gigalixir through the CI (not the CLI first and then the CI) and I got the following error

Screenshot 2021-01-20 at 21 59 40

```bash
Getting current release
  /usr/local/bin/gigalixir releases -a XXXXXXX
  []
  
Error: Cannot read property 'version' of undefined
```
I verified the error by running `gigalixir releases -a XXXX` which returned an empty array as I hadn't done any release previously using the CLI

After making the first release using the CLI I re-run the command and got back
```
$ gigalixir releases -a XXXXXX

[
  {
    "created_at": "2021-01-20T19:40:31.000+00:00",
    "sha": "587fe8dcc70f3557b9912809f3ea745d071b2b43",
    "summary": "Deployed. Set `DATABASE_URL` config var. Set `POOL_SIZE` config var.",
    "version": 1
  }
]
```
So my approach was to set the `currentRelease` number to `0` if no previous release was found because of the checks made (`oldRelease + 1`) to detect readiness, but print an explicit `[NO PREV RELEASE]` 'version' if no previous release was found

Note: also made an unrelated change to remove the `const pods=...` assignment and make the linter happy.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,18 +63,34 @@ async function getCurrentRelease(app) {
     await exec.exec(`gigalixir releases -a ${app}`, [], options);
   });
 
-  const currentRelease = Number(JSON.parse(releasesOutput)[0].version);
+  const releases = JSON.parse(releasesOutput);
+  // first release number in gigalixir is 1
+  const currentRelease = releases.length && Number(releases[0].version) || 0;
```
Comment (mhanberg) [inline index.js:68]:
```suggestion
  const currentRelease = releases.length ? Number(releases[0].version) : 0;
```

File: index.js [hunk 2]
```diff
@@ -64,18 +63,34 @@ async function getCurrentRelease(app) {
     await exec.exec(`gigalixir releases -a ${app}`, [], options);
   });
 
-  const currentRelease = Number(JSON.parse(releasesOutput)[0].version);
+  const releases = JSON.parse(releasesOutput);
+  // first release number in gigalixir is 1
```
Comment (mhanberg) [inline index.js:67]:
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -64,18 +63,34 @@ async function getCurrentRelease(app) {
...
+  const currentRelease = releases.length && Number(releases[0].version) || 0;
 
   return currentRelease;
 }
 
+function printRelease(releaseNumber) {
+  return releaseNumber || "[NO PREV RELEASE]"
+}
```
Comment (mhanberg) [inline index.js:75]:
```suggestion
function formatReleaseMessage(releaseNumber) {
  return releaseNumber
               ? `The current release is ${currentRelease}`
               : "This is the first release";
}
```

File: index.js [hunk 4]
```diff
@@ -92,7 +107,7 @@ async function run() {
     const currentRelease = await core.group("Getting current release", async () => {
       return await getCurrentRelease(gigalixirApp);
     });
-    core.info(`The current release is ${currentRelease}`);
+    core.info(`The current release is ${printRelease(currentRelease)}`);
```
Comment (mhanberg) [inline index.js:110]:
```suggestion
    core.info(formatReleaseMessage(currentRelease));
```

File: index.js [hunk 5]
```diff
@@ -112,18 +127,17 @@ async function run() {
           await exec.exec(`gigalixir ps:migrate -a ${gigalixirApp}`)
         });
       } catch (error) {
-        core.warning(`Migration failed, rolling back to the previous release: ${currentRelease}`);
+        core.warning(`Migration failed, rolling back to the previous release: ${printRelease(currentRelease)}`);
         await core.group("Rolling back", async () => {
           await exec.exec(`gigalixir releases:rollback -a ${gigalixirApp}`)
         });
```
Comment (mhanberg) [inline index.js:133]: This should probably not attempt to roll back if it is the first deploy.

File: index.js [hunk 6]
```diff
@@ -64,18 +63,34 @@ async function getCurrentRelease(app) {
...
+    });
+    const sshPrivateKey = core.getInput('SSH_PRIVATE_KEY', {
+      required: true
+    });
+    const gigalixirApp = core.getInput('GIGALIXIR_APP', {
+      required: true
+    });
+    const migrations = core.getInput('MIGRATIONS', {
+      required: true
+    });
```
Comment (mhanberg) [inline index.js:93]: Can you extract `{required: true}` to a variable called `baseInputOptions` and pass that into these?

e.g., `const gigalixirUsername = core.getInput('GIGALIXIR_USERNAME, baseInputOptions)`

File: index.js [hunk 7]
```diff
@@ -4,7 +4,7 @@ const path = require('path');
 
 function wait(seconds) {
   return new Promise(resolve => {
-    if (typeof(seconds) !== 'number') {
+    if (typeof (seconds) !== 'number') {
```
Comment (mhanberg) [inline index.js:7]:
```suggestion
    if ((typeof seconds) !== 'number') {
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,7 @@
 # Changelog
 
+## v0.4.1
+- Fixed an issue when deploying an app to gigalixir for the first time ever through the CI
```
Comment (mhanberg) [inline CHANGELOG.md:4]:
```suggestion
- Deployment works if the action is making the very first deployment.

```

**Review by mhanberg — CHANGES_REQUESTED** [review]
Thanks for the PR! Please act on or respond to my comments and we can get this merged.

**Comment by bottlenecked** [comment]
Happy to oblige! Let me know if there are more issues.

**Pull request merged**